File: swift/truncate_lines.swift
import Foundation

struct MissingFileName: Error, CustomStringConvertible {
    let description = "Missing file name"
}

do {
    guard CommandLine.argc > 1 else {
        throw MissingFileName()
    }
    let fileName = CommandLine.arguments[1]
    let s = try String(contentsOfFile: fileName)
    let lines = s.components(separatedBy: .newlines) 
    for l in lines {
        print(l.prefix(20))
    }
} catch {
    print(error)
}

File: python/truncate_lines.py
import sys

if len(sys.argv) < 2:
    raise Exception('Missing file name')

fileName = sys.argv[1]
with open(fileName) as f:
    for l in f:
        if len(l) > 10:
            print(l[:10])
        else:
            print(l[:-1])
